internal/mode/static/state/graph/httproute.go

```diff
@@ -384,18 +384,17 @@ func tryToAttachRouteToListeners(
 
 // findAttachableListeners returns a list of attachable listeners and whether the listener exists for a non-empty
 // sectionName.
-func findAttachableListeners(sectionName string, listeners map[string]*Listener) ([]*Listener, bool) {
+func findAttachableListeners(sectionName string, listeners []*Listener) ([]*Listener, bool) {
 	if sectionName != "" {
-		l, exists := listeners[sectionName]
-		if !exists {
-			return nil, false
-		}
-
-		if l.Attachable {
-			return []*Listener{l}, true
+		for _, l := range listeners {
```

Comment:
Important part: This is the only one that performs the lookup operation for the map `listeners`. Although iterating through the listeners slice increases the time complexity from O(1) to O(n), this approach eliminates the need for sorting in prepareGatewayStatus, which has a time complexity of O(n log n). Hence, there is no performance issue here.

Reply:
I'll add that `n=64` in this case since the API has a max of 64 listeners per Gateway, so I'm not worried about the performance of this lookup.

Reply:
Cool, this is good news!